pilot/pkg/serviceregistry/serviceentry/servicediscovery.go

```diff
@@ -179,7 +199,7 @@ func (s *ServiceEntryStore) workloadEntryHandler(old, curr config.Config, event
 		event = model.EventDelete
 	}
 
-	wi := s.convertWorkloadEntryToWorkloadInstance(curr, s.Cluster())
+	wi := s.convertWorkloadEntryToWorkloadInstance(curr, currLabels, s.Cluster())
```

Comment:
What about `convertWorkloadEntryToServiceInstances`?

Reply:
good catch, need to fix too